src/lib.rs

```diff
@@ -0,0 +1,339 @@
+mod network;
+
+#[allow(dead_code, unused_imports)]
+
+pub type Meters = f32;
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum End {
+    Front,
+    Back,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum Direction {
+    Forward,
+    Backward,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum Directions {
+    Forward,
+    Backward,
+    BothWays,
+    Alternating,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum Side {
+    Left,
+    Right,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum SideDirection {
+    Leftward,
+    Rightward,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum LaneSide {
+    Inside,
+    Outside,
+}
+
+pub enum LaneDirection {
+    Inward,
+    Outward,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum DrivingSide {
+    LHT,
+    RHT,
+}
+
+use Direction::*;
```

Comment:
What does importing an enum like this mean? I don't think I've ever seen just `LHT` used in code to mean `DrivingSide::LHT`, for example

Reply:
You are on the right track, it `use`s each option in that scope, so that you could quote `LHT` to mean `DrivingSide::LHT`.

It's like a prelude that lets you say things like `Some(Inside)`.

Reply:
iirc, that is why C++ removed un-namespaces enums from C...